sale_operating_unit/models/sale_order.py

```diff
@@ -0,0 +1,79 @@
+# © 2019 ForgeFlow S.L.
+# - Jordi Ballester Alomar
+# © 2019 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model
+    def _default_operating_unit(self):
+        team = self.env["crm.team"]._get_default_team_id()
+        if team.operating_unit_id:
+            return team.operating_unit_id
+        return self.env.user.default_operating_unit_id
+
+    operating_unit_id = fields.Many2one(
+        comodel_name="operating.unit",
+        string="Operating Unit",
+        default=_default_operating_unit,
+        readonly=True,
+        states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
+    )
+
+    @api.onchange("team_id")
+    def onchange_team_id(self):
+        if self.team_id:
+            self.operating_unit_id = self.team_id.operating_unit_id
+
+    @api.onchange("operating_unit_id")
+    def onchange_operating_unit_id(self):
+        if self.team_id and self.team_id.operating_unit_id != self.operating_unit_id:
+            self.team_id = False
```

Comment:
```suggestion
            self.team_id = False
        if self.operating_unit_id:
            self.team_id = self.env["crm.team"].search(
                [("operating_unit_id", "=", self.operating_unit_id.id)], limit=1
            )
```

To set a team when the operating unit is changed